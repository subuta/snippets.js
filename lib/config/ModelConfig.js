import check from 'lib/_utils/apiCheck'

const ModelConfigShape = check.shape({
  schema: check.object
})

// eg:
// config = {
//   book: {
//     schema: {}
//   }
// }
export const validateModels = function (config) {
  return check.throw(check.objectOf(ModelConfigShape), config)
}

// eg:
// config = {
//   schema: {}
// }
export const validateModel = function (config) {
  return check.throw(ModelConfigShape, config)
}
