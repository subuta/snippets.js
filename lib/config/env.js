import { build, format, snippets as s } from 'bld.js'
import _ from 'lodash'

export default (imports = []) => {
  const vars = _.zipObject(imports, imports)

  return build`
    // load .env first.
    require('dotenv').config()
    
    const ${s.raw(vars)} = process.env
    
    export ${s.raw(vars)}
    
    export default ${s.raw(vars)}
  `
}
