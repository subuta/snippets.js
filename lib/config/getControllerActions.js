import _ from 'lodash'

export default (config = {}) => {
  const {
    only, // the routes will generated.
    except  // the routes will ignored.
  } = config

  const routes = [
    'index',
    'show',
    'create',
    'update',
    'destroy'
  ]

  if (only) {
    return _.intersection(routes, only)
  } else if (except) {
    return _.difference(routes, except)
  }

  return routes
}
