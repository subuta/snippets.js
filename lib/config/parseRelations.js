import _ from 'lodash'
import pluralize from 'pluralize'

export default (config = {}) => {
  const { schema } = config

  const relations = _.transform(schema.relations, (result, value, key) => {
    let relatedModel = null;
    if (value.belongsTo) {
      relatedModel = pluralize.singular(_.toLower(value.belongsTo))
    } else if (value.hasOne) {
      relatedModel = pluralize.singular(_.toLower(value.hasOne))
    } else if (value.hasMany) {
      relatedModel = pluralize(_.toLower(value.hasMany))
    } else if (value.hasAndBelongsToMany) {
      relatedModel = pluralize(_.toLower(value.hasAndBelongsToMany))
    }
    result[key] = relatedModel
  }, {})

  return relations
}
