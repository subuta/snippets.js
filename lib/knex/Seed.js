import { build, format, snippets as s } from 'bld.js'
import _ from 'lodash'

import {
  generateSeedsWithRelation,
  generateJunctionSeeds
} from 'lib/_utils/generateSeed'

export default (props) => {
  let {model, modelsConfig} = props

  const modelConfig = modelsConfig[model]
  const {isJunction, schema} = modelConfig
  const tableName = s.tableName(schema.tableName)

  let seeds = []
  if (isJunction) {
    const maxCounts = _.reduce(_.toPairs(schema.relations), (acc, [relationModel, relation]) => {
      const relationModelConfig = modelsConfig[relationModel]
      return _.max([acc, relationModelConfig.seeds])
    }, 0)
    seeds = generateJunctionSeeds(model, modelsConfig, maxCounts)
  } else {
    seeds = generateSeedsWithRelation(model, modelsConfig, modelConfig.seeds)
  }

  return build`
    exports.seed = async (knex) => {
      // Deletes ALL and insert entries.
      await knex('${tableName}').del()
    
      // https://github.com/tgriesser/knex/issues/54
      await knex('${tableName}').insert(${s.stringify(seeds)})
    }
  `
}
