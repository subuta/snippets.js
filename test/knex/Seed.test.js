/* global expect, describe, it */

import { build, format, snippets as s } from 'bld.js'

import Seed from 'lib/knex/Seed'

import { Routes as RoutesConfig, Models as ModelsConfig } from 'test/fixtures/config'

describe('knex/Entrypoint', () => {
  it('should create knex Entrypoint', () => {
    expect(format(Seed({
      model: 'user',
      routeConfig: RoutesConfig.user,
      modelConfig: ModelsConfig.user
    }))).toMatchSnapshot()
  })
})
