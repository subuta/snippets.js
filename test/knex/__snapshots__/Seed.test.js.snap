// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`knex/Seed should create knex seed 1`] = `
"exports.seed = async (knex) => {
  // Deletes ALL and insert entries.
  await knex('users').del()

  // https://github.com/tgriesser/knex/issues/54
  await knex('users').insert([
    {
      id: 54551,
      auth0Id: '4be76d72-37c6-4af4-9b6d-5cf3cb62d63a',
      locale: 'en_au_ocker',
      nickname: 'Juana5',
      status: 'New Israeli Sheqel',
      avatar:
        'https://s3.amazonaws.com/uifaces/faces/twitter/devankoshal/128.jpg'
    },
    {
      id: 15599,
      auth0Id: 'e5ad378c-913f-4366-bb72-507428a10360',
      locale: 'en_BORK',
      nickname: 'Shanel.Barton',
      status: 'web-enabled',
      avatar:
        'https://s3.amazonaws.com/uifaces/faces/twitter/antongenkin/128.jpg'
    },
    {
      id: 34490,
      auth0Id: 'c5d5096e-8b5e-4c16-9cc6-f15cf8fdcef5',
      locale: 'en_au_ocker',
      nickname: 'Marisol66',
      status: 'Mobility Missouri Administrator',
      avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/dgajjar/128.jpg'
    }
  ])
}
"
`;

exports[`knex/Seed should create knex seed for channel 1`] = `
"exports.seed = async (knex) => {
  // Deletes ALL and insert entries.
  await knex('channels').del()

  // https://github.com/tgriesser/knex/issues/54
  await knex('channels').insert([
    {id: 68264, name: 'convergence Administrator Clothing'},
    {id: 36069, name: 'Baby Lead solutions'},
    {id: 63669, name: 'functionalities'}
  ])
}
"
`;

exports[`knex/Seed should create knex seed for comment 1`] = `
"exports.seed = async (knex) => {
  // Deletes ALL and insert entries.
  await knex('comments').del()

  // https://github.com/tgriesser/knex/issues/54
  await knex('comments').insert([
    {
      id: 57447,
      text:
        'Ut amet molestiae aliquam numquam ad ut et. Et ut perspiciatis fugiat expedita fuga hic odio. Ducimus cumque aut harum in perferendis veritatis tempora aut. Facilis officia sed. Eum excepturi officia ipsum sunt voluptatem eius. Rerum qui doloremque ex velit porro animi odio saepe sunt.',
      channelId: 68264,
      commentedById: 54551,
      attachmentId: '28d15c5a-a70c-48e4-9772-bc910f421907'
    },
    {
      id: 33984,
      text:
        'Id optio rem culpa minus. Nostrum aut et. Aliquam repellat est reprehenderit.',
      channelId: 36069,
      commentedById: 15599,
      attachmentId: 'abfbc020-04b2-4d70-a1ed-aa85e702e181'
    },
    {
      id: 46230,
      text:
        'Facilis cupiditate quo nemo hic. Rem nesciunt aliquid aut qui architecto saepe enim laborum. Quia dolor consequuntur et dolorem tenetur totam nisi asperiores.',
      channelId: 63669,
      commentedById: 34490,
      attachmentId: '62fc4caf-224d-4658-949e-b68307f974d2'
    }
  ])
}
"
`;

exports[`knex/Seed should create knex seed for junction table 1`] = `
"exports.seed = async (knex) => {
  // Deletes ALL and insert entries.
  await knex('articles_tags').del()

  // https://github.com/tgriesser/knex/issues/54
  await knex('articles_tags').insert([
    {id: 48239, articleId: 52697, tagId: 82337},
    {id: 52023, articleId: 14400, tagId: 43149},
    {id: 37083, articleId: 29655, tagId: 3838}
  ])
}
"
`;
